examples/call-runtime/lib.rs

```diff
@@ -0,0 +1,222 @@
+    #[ink(storage)]
+    #[derive(Default)]
+    pub struct RuntimeCaller;
+
+    impl RuntimeCaller {
+        /// The constructor is `payable`, so that during instantiation it can be given some tokens
+        /// that will be further transferred with `transfer_through_runtime` message.
+        #[ink(constructor, payable)]
+        pub fn new() -> Self {
+            Default::default()
+        }
+
+        /// Tries to transfer `value` from the contract's balance to `receiver`.
+        ///
+        /// Fails if:
+        ///  - called in the off-chain environment
+        ///  - the chain doesn't allow `call-runtime` API (`UnsafeUnstableInterface` is turned off)
+        ///  - the chain forbids contracts to call `Balances::transfer` (`CallFilter` is too
+        ///    restrictive)
+        ///  - after the transfer, `receiver` doesn't have at least existential deposit  
+        ///  - the contract doesn't have enough balance
+        #[ink(message)]
+        pub fn transfer_through_runtime(
+            &mut self,
+            receiver: AccountId,
+            value: Balance,
+        ) -> Result<(), ()> {
+            self.env()
+                .call_runtime(&RuntimeCall::Balances(BalancesCall::Transfer {
+                    dest: receiver.into(),
+                    value,
+                }))
+                .map_err(|_| ())
+        }
+    }
+
+    #[cfg(test)]
+    mod tests {
+        use super::*;
+
+        use ink::env::{
+            test::default_accounts,
+            DefaultEnvironment,
+        };
+
+        #[ink::test]
+        #[should_panic(
+            expected = "off-chain environment does not support `call_runtime`"
+        )]
+        fn cannot_call_runtime_off_chain() {
+            let mut contract = RuntimeCaller::new();
+            let _call_res = contract.transfer_through_runtime(
+                default_accounts::<DefaultEnvironment>().bob,
+                10,
+            );
+        }
+    }
+
+    #[cfg(all(test, feature = "e2e-tests"))]
+    mod e2e_tests {
+        use super::*;
+
+        use ink::{
+            env::{
+                test::default_accounts,
+                DefaultEnvironment,
+            },
+            primitives::AccountId,
+        };
+        use ink_e2e::build_message;
+
+        type E2EResult<T> = Result<T, Box<dyn std::error::Error>>;
+
+        /// The contract will be given 1000 tokens during instantiation.
+        const CONTRACT_BALANCE: Balance = 1_000_000_000_000_000;
+        /// The receiver will get enough funds to have the required existential deposit.
+        ///
+        /// If your chain has this threshold higher, increase the transfer value.
+        const TRANSFER_VALUE: Balance = 1_000_000_000;
+
+        #[ink_e2e::test]
+        #[ignore = "Requires that the pallet contract is configured with:\
+            - `CallFilter` allowing for a transfer, e.g. `frame_support::traits::Everything`,\
+            - `UnsafeUnstableInterface = ConstBool<true>`"]
```

Comment:
So I'd add the another feature flag to the contract and enable the test through that, instead of simply ignoring it

Reply:
what would be such a feature? `run-e2e`? `work`?

Reply:
added `permissive-node` feature